pkg/tfbridge/x/token_test.go

```diff
@@ -169,6 +169,112 @@ func TestIgnored(t *testing.T) {
 	}, info.Resources)
 }
 
+func TestTokensInferredModules(t *testing.T) {
+	tests := []struct {
+		name            string
+		resourceMapping map[string]string
+		opts            *InferredModulesOpts
+	}{
+		{
+			name: "non-overlapping mapping",
+			resourceMapping: map[string]string{
+				"pkg_foo_bar":             "index:FooBar",
+				"pkg_fizz_buzz":           "index:FizzBuzz",
+				"pkg_resource":            "index:Resource",
+				"pkg_very_long_name":      "index:VeryLongName",
+				"pkg_very_very_long_name": "index:VeryVeryLongName",
+			},
+		},
+		{
+			name: "detect a simple module",
+			resourceMapping: map[string]string{
+				"pkg_hello_world":   "hello:World",
+				"pkg_hello_pulumi":  "hello:Pulumi",
+				"pkg_hello":         "hello:Hello",
+				"pkg_goodbye_folks": "index:GoodbyeFolks",
+				"pkg_hi":            "index:Hi",
+			},
+			opts: &InferredModulesOpts{
+				MinimumModuleSize: 2,
+			},
+		},
+		{
+			name: "nested modules",
+			resourceMapping: map[string]string{
+				"pkg_mod_r1":     "mod:R1",
+				"pkg_mod_r2":     "mod:R2",
+				"pkg_mod_r3":     "mod:R3",
+				"pkg_mod_r4":     "mod:R4",
+				"pkg_mod_sub_r1": "modSub:R1",
+				"pkg_mod_sub_r2": "modSub:R2",
+				"pkg_mod_sub_r3": "modSub:R3",
+				"pkg_mod_sub_r4": "modSub:R4",
+				"pkg_mod_not_r1": "mod:NotR1",
+				"pkg_mod_not_r2": "mod:NotR2",
+			},
+			opts: &InferredModulesOpts{
+				TfPkgPrefix:          "pkg_",
+				MinimumModuleSize:    3,
+				MimimumSubmoduleSize: 4,
```

Comment:
How is MimimumSubmoduleSize: 4 in play in this example?

Reply:
I have annotated that test. Here are is the new object:
```go
		{
			name: "nested modules",
			resourceMapping: map[string]string{
				"pkg_mod_r1":     "mod:R1",
				"pkg_mod_r2":     "mod:R2",
				"pkg_mod_r3":     "mod:R3",
				"pkg_mod_r4":     "mod:R4",
				"pkg_mod_sub_r1": "modSub:R1",
				"pkg_mod_sub_r2": "modSub:R2",
				"pkg_mod_sub_r3": "modSub:R3",
				"pkg_mod_sub_r4": "modSub:R4",
				"pkg_mod_not_r1": "mod:NotR1",
				"pkg_mod_not_r2": "mod:NotR2",
			},
			opts: &InferredModulesOpts{
				TfPkgPrefix: "pkg_",
				// We set the minimum module size to 4. This ensures that
				// `pkg_mod` is picked up as a module.
				MinimumModuleSize: 4,
				// We set the MimimumSubmoduleSize to 3, ensuring that
				// `pkg_mod_sub_*` is is given its own `modSub` module (4
				// elements), while `pkg_mod_not_*` is put in the `mod`
				// module, since `pkg_mod_not` only has 2 elements.
				MimimumSubmoduleSize: 3,
			},
		}
```